src/compas_timber/ghpython/components/CT_Joint_Rule_Category/code.py

```diff
@@ -3,7 +3,64 @@
 from compas_timber.ghpython import CategoryRule
 
 
-class CategoryJointRule(component):
-    def RunScript(self, JointOptions, CatA, CatB):
-        if JointOptions and CatA and CatB:
-            return CategoryRule(JointOptions.type, CatA, CatB, **JointOptions.kwargs)
+def add_param(name, io):
+    assert io in ("Output", "Input")
+    params = [param.NickName for param in getattr(ghenv.Component.Params, io)]
+    if name not in params:
+        param = Grasshopper.Kernel.Parameters.Param_GenericObject()
+        param.NickName = name + " category"
+        param.Name = name
+        param.Description = name
+        param.Access = Grasshopper.Kernel.GH_ParamAccess.item
+        param.Optional = True
+        index = getattr(ghenv.Component.Params, io).Count
+        registers = dict(Input="RegisterInputParam", Output="RegisterOutputParam")
+        getattr(ghenv.Component.Params, registers[io])(param, index)
+        ghenv.Component.Params.OnParametersChanged()
+
+def clear_params():
+    while len(ghenv.Component.Params.Input) > 1:
+        ghenv.Component.Params.UnregisterInputParameter(
+            ghenv.Component.Params.Input[len(ghenv.Component.Params.Input) - 1]
+        )
+    ghenv.Component.Params.OnParametersChanged()
+    ghenv.Component.ExpireSolution(False)
```

Comment:
There's a module called `ghcomponent_helpers.py` under `ghpython, maybe move these functions there? that way you can reuse them in the other components instead of redefining